### ISSUE TITLE
`summary` with `as.Date` converts date to integer

### ISSUE THREAD

**Issue by mrdwab** [Rapporter/pander#237]
`as.Date` objects seem to require an `as.character` to suppress printing as integers when using `summary` in Pander 0.6.0

Quick demo:

```r
set.seed(1); df <- data.frame(id = 1:10, date = Sys.Date() + sample(-50:50, 10, TRUE))
pander(df)
```

Output is OK:

```
---------------
 id     date   
---- ----------
 1   2015-12-04

 2   2015-12-15

 3   2016-01-04

 4   2016-02-07

 5   2015-11-28

 6   2016-02-06

 7   2016-02-11

 8   2016-01-13

 9   2016-01-10

 10  2015-11-14
---------------
```

But:

```r
pander(summary(df$date))
```

Output is converted to integer:

```
-------------------------------------------------
 Min.   1st Qu.   Median   Mean   3rd Qu.   Max. 
------ --------- -------- ------ --------- ------
16753    16776    16807   16802    16831   16842 
-------------------------------------------------
```

**Comment by mrdwab** [comment]
See: http://stackoverflow.com/questions/34493025/r-knitr-pander-how-to-nicely-format-summary-of-dates

**Comment by daroczig** [comment]
Thanks for reporting this issues, which happens due to the behavior of `as.matrix` we used to standardize different tabular objects in 0.6. See eg:

```r
> summary(df$date)
        Min.      1st Qu.       Median         Mean      3rd Qu.         Max. 
"2015-11-14" "2015-12-06" "2016-01-07" "2016-01-02" "2016-01-31" "2016-02-11" 
```

versus

```r
> as.matrix(summary(df$date))
            [,1]
Min.    16753.00
1st Qu. 16775.75
Median  16807.00
Mean    16802.10
3rd Qu. 16831.00
Max.    16842.00
```

I'll think about the potential solutions for this.

**Comment by daroczig** [comment]
It seems to me that converting all cells to character after taking care of the numeric values should work at https://github.com/Rapporter/pander/blob/master/R/pandoc.R#L952 -- will do some testing and push a fix in the next few days.

**Comment by daroczig** [comment]
The above commit should do the trick:

```r
> pander(summary(df$date))

-----------------------------------------------------------------
   Min.     1st Qu.     Median      Mean     3rd Qu.      Max.   
---------- ---------- ---------- ---------- ---------- ----------
2015-11-14 2015-12-06 2016-01-07 2016-01-02 2016-01-31 2016-02-11
-----------------------------------------------------------------
```

Please let me know how it works.

**Comment by mrdwab** [comment]
@daroczig Looks good! Extremely rapid response, once again :-)

**Comment by rgayler** [comment]
Thanks @mrdwab and @daroczig. I confirm that works for me.

**Comment by daroczig** [comment]
Thanks for the quick feedback, much appreciated.

**Comment by rgayler** [comment]
I may have spoken too soon. It appears the fix has introduced a new problem. I am pretty sure the last example rendered OK before the fix (apart from the dates being formatted as integers).

I'm not sure what the protocol is for reopening an issue on GitHub, so I have attached it as a comment to issue #237:

```
> summary(mq_cc$application_decision_date)
        Min.      1st Qu.       Median         Mean      3rd Qu.         Max.         NA's 
"2014-05-01" "2015-02-13" "2015-06-18" "2015-05-09" "2015-09-11" "2015-11-21"       "1994" 

> summary(mq_cc$application_decision_date) %>% as.character() %>% pander::pander()

------------------------------------------------------------------------
   Min.     1st Qu.     Median      Mean     3rd Qu.      Max.     NA's 
---------- ---------- ---------- ---------- ---------- ---------- ------
2014-05-01 2015-02-13 2015-06-18 2015-05-09 2015-09-11 2015-11-21  1994 
------------------------------------------------------------------------

> summary(mq_cc$application_decision_date) %>% pander::pander()
Error in `colnames<-`(`*tmp*`, value = c("Min.", "1st Qu.", "Median",  : 
  length of 'dimnames' [2] not equal to array extent
```

**Comment by daroczig** [comment]
@rgayler can you please post a reproducible example of the problem? I don't know that is `mq_cc`.

And did you install `pander` from GitHub via eg `devtools::install_github('Rapporter/pander')`? 

Here `summary` seems to be OK with dates now:

```r
> pander::pander(summary(as.Date('2015-01-01') + 1:100))

-----------------------------------------------------------------
   Min.     1st Qu.     Median      Mean     3rd Qu.      Max.   
---------- ---------- ---------- ---------- ---------- ----------
2015-01-02 2015-01-26 2015-02-20 2015-02-20 2015-03-17 2015-04-11
-----------------------------------------------------------------

> pander::pander(as.character(summary(as.Date('2015-01-01') + 1:100)))

-----------------------------------------------------------------
   Min.     1st Qu.     Median      Mean     3rd Qu.      Max.   
---------- ---------- ---------- ---------- ---------- ----------
2015-01-02 2015-01-26 2015-02-20 2015-02-20 2015-03-17 2015-04-11
-----------------------------------------------------------------
```

If you can also share your `devtools::session_info()`, that would be also very helpful.

**Comment by rgayler** [comment]
Hi Gergely,

I have submitted a reproducible example as a new issue (#238). Sorry if
that's the wrong way to do it - I don't have any GitHub expertise.


> And did you install pander from GitHub via eg
devtools::install_github('Rapporter/pander')?

Yes, I did install from github, but I am running packrat at the moment, so
I had to install with packrat::install_github('Rapporter/pander')

> share your devtools::session_info()

I don't have devtools installed right now, so I put the output from
utils::sessionInfo in the reproducible example. I am installing devtools in
the background now.

Ross
